solver/src/http_solver.rs

```diff
@@ -190,6 +203,16 @@ impl HttpSolver {
             .get_token_infos(addresses.as_slice())
             .await;
 
+        let price_estimates: HashMap<H160, Result<BigRational>> = addresses
+            .iter()
+            .cloned()
+            .zip(
+                self.price_estimator
+                    .estimate_prices(addresses.as_slice(), addresses[0])
```

Comment:
`addresses[0]` will panic if there are 0 addresses in total. Just saying because I'm not sure if we guarantee that there is at least one address at this point.

Reply:
I believe it must because of [this](https://github.com/gnosis/gp-v2-services/blob/112c13ac2b04d6401d5092c5f495148627cc2d14/solver/src/http_solver.rs#L333-L335), right?

Reply:
Yes, looks right